docs/pod-template.md

```diff
@@ -0,0 +1,77 @@
+# Pod Template
+
+![alpha](assets/alpha.svg)
+
+> v3.1 and after
+
+A pod templates is similar to a normal container or script template, but allows you to specify multiple containers to
+run within the pod.
+
+Because you have multiple containers within a pod, they will be scheduled on the same host. You can use cheap and fast
+empty-dir volumes instead of persistent volume claims to share data between steps.
+
+For example, if you have nested DAGs, and the nested DAGs would benefit from a shared workspace.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: containers-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      volumes:
+        - name: workpsace
+          emptyDir: { }
+      pod:
+        volumeMounts:
+          - mountPath: /workspace
+            name: workspace
+        graph:
+          - name: a
+            image: argoproj/argosay:v2
+            args: [ echo, hello, /mnt/share/message ]
+          - name: b
+            image: argoproj/argosay:v2
+          - name: main
+            image: argoproj/argosay:v2
+            args: [ cat, /workpsace/message ]
+            dependencies:
+              - a
+              - b
+      outputs:
+        parameters:
+          - name: message
+            valueFrom:
+              path: /workpsace/message
+```
+
+There are a couple of caveats:
+
+1. You must use the [Emissary Executor](workflow-executors.md#emissary-emissary).
```

Comment:
Is it technically infeasible to support other types executors? If not, we can reword to something like `currently, only support...` so others know what to expect.